### DISCUSSION TITLE
General

### DISCUSSION THREAD

**Discussion by adamspd** [adamspd/django-appointment#167] [General]
Let's discuss here if need be (for other matters not related to a specific issue or PR).
@deronnax

**Comment by 254kayjay** [comment]
So how does this site work?

**Reply by adamspd** [reply]
Hi @254kayjay, what do you mean ? Are you talking about django-appointment ? 
If yes, then it's not a site, it's a package (Django app) that you install in your personal project that allows your client to make appointment and you to manage them.

You can test it by following [these instructions](https://github.com/adamspd/django-appointment?tab=readme-ov-file#docker-support-).

**Comment by kennyhei** [comment]
Hi, this is really cool project. I have implemented my own booking service and now a year and a half later I discovered this one. Seems like it does not exactly fit for my current needs but I will definitely use this as an inspiration. Here are some feature request ideas:

- Support additional services for the "base" service
    - Example: Hair salon might offer haircut + additional scalp massage for small extra fee
- Flexible slot duration where customer can select how long the service is (I think you have created ticket about this?)
    - Example: Car rental service where customer can select for how many days they would like to rent the vehicle
- Support discount codes / coupons
    - Percentage discounts (-10% off)
    - Fixed discounts (-10$ off)
    - Fixed price discounts (let's say the service costs 500$ but with discount code it is always 100$)

**Reply by adamspd** [reply]
Hi @kennyhei, thank you for the ideas :)
I will try to implement them...
